responders/src/API/EMBC.Responders.API/Controllers/ProfileController.cs

```diff
@@ -108,15 +131,23 @@ public async Task<ActionResult> Update(UpdateUserProfileRequest request)
         /// <returns>Ok when successful</returns>
         [HttpPost("agreement")]
         [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
         public async Task<IActionResult> SignAgreement()
         {
-            var memberId = User.FindFirstValue(ClaimTypes.Sid);
-            var teamId = User.FindFirstValue("user_team");
+            var userName = User.FindFirstValue(ClaimTypes.Upn).Split('@')[0];
+            var userId = User.FindFirstValue(ClaimTypes.Sid);
+            var sourceSystem = User.FindFirstValue("identity_source");
 
             // Get the current user
-            var reply = await messagingClient.Send(new TeamMembersQueryCommand { TeamId = teamId, MemberId = memberId, IncludeActiveUsersOnly = true });
+            var reply = await messagingClient.Send(new TeamMembersQueryCommand { UserName = userName, IncludeActiveUsersOnly = true });
```

Comment:
this is fine to use member id - the user name was already successfully authenticated